src/runner/test-run/index.js

```diff
@@ -171,6 +197,33 @@ ServiceMessages[COMMAND.nativeDialogsInfoSet] = function (msg) {
     }
 };
 
-ServiceMessages[COMMAND.takeScreenshot] = function () {
-    //TODO:
+ServiceMessages[COMMAND.takeScreenshot] = async function (msg) {
+    await ensureDir(this.screenshotPath);
+
+    var pageUrl      = msg.pageUrl;
+    var stepName     = msg.stepName;
+    var isFailedStep = msg.isFailedStep;
+
+    var filePath = msg.filePath ?
+                   path.join(path.dirname(this.test.fixture.path), msg.filePath) :
+                   `${this.screenshotPath}\\${isFailedStep ? 'errors\\' : ''}${stepName.replace(/\s/g, '_') ||
```

Comment:
Btw, use `path.join` instead for parts of the path, since separators may differ on different OSes